ios/SnapyrRnSdk.m

```diff
@@ -13,20 +13,23 @@ @implementation SnapyrRnSdk
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)
 {
-  SnapyrSDKConfiguration *configuration = [SnapyrSDKConfiguration configurationWithWriteKey:key];
-  if ([_options objectForKey:@"trackApplicationLifecycleEvents"]) {
-    configuration.trackApplicationLifecycleEvents = YES; // Enable this to record certain application events automatically
-  }
-  if ([_options objectForKey:@"recordScreenViews"]) {
-    configuration.recordScreenViews = YES; // Enable this to record screen views automatically
-  }
-  if ([_options objectForKey:@"enableDevMode"]) {
-    configuration.enableDevEnvironment = YES; // Test against a Snapyr dev environment *internal only*
-  }
-
-  [SnapyrSDK setupWithConfiguration:configuration];
-
-  resolve(key);
+    SnapyrSDKConfiguration *configuration = [SnapyrSDKConfiguration configurationWithWriteKey:key];
+    if ([_options objectForKey:@"trackApplicationLifecycleEvents"]) {
+        configuration.trackApplicationLifecycleEvents = YES; // Enable this to record certain application events automatically
+    }
+    if ([_options objectForKey:@"recordScreenViews"]) {
+        configuration.recordScreenViews = YES; // Enable this to record screen views automatically
+    }
+    if ([_options objectForKey:@"enableDevMode"]) {
+        configuration.enableDevEnvironment = YES; // Test against a Snapyr dev environment *internal only*
+    }
+    
+    // makes every event flush to network immediately
+    configuration.flushAt = 1;
```

Comment:
Appreciate comment, but still not sure exactly what that means? What's the default value/behavior?

Reply:
By default, the native SDKs queue up events (track/identify) and periodically "flush" them (i.e. send to Snapyr).

The default behavior is to flush every 30 seconds, or whenever the queue size reaches 20. 

Setting this to 1 is equivalent to "auto flushing" right after every track. This was already being done on the Android side of the React Native SDK but not here, so this update brings it to parity... we may want to change this later, but for now it helps to prevent issues with KPIs on push notifications from "going missing" if the app is in the background and doesn't flush the queue in time.

Reply:
Got it, thanks for the explanation, makes sense!